src/tasks/processBulkUpload.ts

```diff
@@ -14,4 +96,15 @@ export const processBulkUpload = async (
     );
   }
   helpers.logger.debug(`Started processBulkUpload Job for Bulk Upload ID ${payload.bulkUploadId}`);
+  const bulkUpload = await loadBulkUpload(payload.bulkUploadId);
+  let bulkUploadFile: FileResult;
+  try {
+    await updateBulkUploadStatus(bulkUpload.id, BulkUploadStatus.IN_PROGRESS);
+    bulkUploadFile = await downloadUrlToTemporaryStorage(bulkUpload.sourceUrl);
+  } catch (err) {
+    await updateBulkUploadStatus(bulkUpload.id, BulkUploadStatus.FAILED);
+    return;
```

Comment:
My understanding is that the job gets marked as complete if the promise resolves, and marks as failed if the promise does not resolve.

I'm open to adding retry cases, though that does become a more complex system and it might make sense to start off erring on the side of *not* retrying unless there's an obvious reason (I think you mention one below).

Part of that is that I don't want to think about tuning the retry logic just yet (I suspect doing that with fidelity will be at least an hour of research and documentation, and it is probably most important to build out core functionality before adding that enhancement).